dask/dataframe/core.py

```diff
@@ -2965,3 +2982,11 @@ def drop_columns(df, columns, dtype):
     df = df.drop(columns, axis=1)
     df.columns = df.columns.astype(dtype)
     return df
+
+
+def safe_head(head, df, n):
+    r = head(df, n=n)
+    if len(r) != n:
+        warnings.warn("Insufficient rows in head, expected {} and got {}. "
+                      "Try providing greater npartitions".format(n, len(r)))
```

Comment:
Since `pandas.DataFrame.head`/`toolz.take` don't throw a warning/error if the number of elements is less than the requested number, perhaps we shouldn't either if `npartitions=-1`?

```python
In [11]: df = pd.DataFrame({'a': [1, 2, 3, 4, 5]})

In [12]: df.head(10)
Out[12]:
   a
0  1
1  2
2  3
3  4
4  5
```